### PR TITLE
feat: linting

### PR DESCRIPTION
**Pull request by Aslemammad**
- adding eslint as a pre-commit hook.
- adding eslint to ci + typecheck

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,29 +1,36 @@
...
+    "lint:fix": "eslint . --fix",
+    "test:unit": "vitest"
+  },
   "dependencies": {
     "@create-figma-plugin/utilities": "2.6.1",
     "code-block-writer": "12.0.0",
     "figx": "0.1.3"
   },
   "devDependencies": {
+    "@antfu/eslint-config": "^1.1.0",
```
Comment (NoriSte) [inline package.json:23]: **Suggestion**: I'd rather pin exact versions for all the new deps

File: package.json [hunk 2]
```diff
@@ -1,29 +1,36 @@
 {
+  "type": "module",
```
Comment (NoriSte) [inline package.json:2]: **Question**: Is it needed? If I run `build` I get
> error esbuild error
    require() of ES Module /Users/noriste/Sites/noriste/figma-to-xstate/build-figma-plugin.main.js from /Users/noriste/Sites/noriste/figma-to-xstate/node_modules/.pnpm/import-fresh@3.3.0/node_modules/import-fresh/index.js not supported.
    build-figma-plugin.main.js is treated as an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which declares all .js files in that package scope as ES modules.
    Instead rename build-figma-plugin.main.js to end in .cjs, change the requiring code to use dynamic import() which is available in all CommonJS modules, or change "type": "module" to "type": "commonjs" in /Users/noriste/Sites/noriste/figma-to-xstate/package.json to treat all .js files as CommonJS (using .mjs for all ES modules instead).

**Review by NoriSte — APPROVED** [review]
**FYI**: I rebased the branch over `main` after merging #6 

**Proposal**: What about also adding [the VSCode configuration](https://github.com/antfu/eslint-config#vs-code-support-auto-fix) to the project?

**Praise**: Thank you so much for the PR 😍😍😍

**Comment by NoriSte** [comment]
**Suggestion**: Remove the no-more relevant points from the [To do list](https://github.com/NoriSte/figma-to-xstate/blob/main/CONTRIBUTING.md#to-do-list) 😊

File: package.json (hunk 2 above)
Comment (Aslemammad) [reply package.json:2]: Yes, it's a recommendation from the eslint plugin. And it's better to keep things ESM.

**Review by NoriSte — APPROVED** [review]
Thank you ♥️

**Pull request merged**